ziptastic-java initial pull request.

bfranco94 opened the pull request:
Ziptastic Object Created
Constructor Class Taking a String input (API KEY) Implemented. 
Code implementing Forward Geocoding completed.

bfranco94 committed:
Add files via upload

bfranco94 committed:
Added Documentation to README.md

Added documentation for initial implementation of the Ziptastic object classes and constructor.

bfranco94 committed:
Ziptastic.java reformatted into an object class.
Constructor Created
Forward Geocoding Method Implemented

daspecster commented on README.md:
```diff
@@ -1,3 +1,41 @@
-# Offcial Ziptastic Java Library
-
-## Coming soon!
+# **Offcial Ziptastic Java Library**
```
Nice! So I was thinking, and what if we put something like **Offcial Ziptastic Java Library (BETA)** until we get this under test? That way we could soft release it(merge) and then continue working.

bfranco94 replied:
Would, Official Ziptastic Java Library (ALPHA) work just as well? I feel as if using BETA indicates more functionality/testing then is present at the moment.

daspecster replied:
👍

daspecster commented on Ziptastic/src/main/java/Ziptastic_Object/Ziptastic.java:
```diff
@@ -0,0 +1,54 @@
+    }
+    
+    
+    public static String ForwardGeocoding(int zipcode) throws IOException{
```
Just a couple thoughts here,
1. Should we call it ForwardGeocode()? Or even GetFromPostalCode() assuming GetFromCoordinates() later on?
2. Can we make another signature so it accepts a string as a `zipcode` as well as an integer?

bfranco94 replied:
1. ForwardGeocode() would probably be a better name. I feel as if GetFromPostalCode() is a little lengthy, but that may just be a matter of personal preference. But at the same time, it would be easier for someone that doesn't fully understand the geocoding terminology to pickup on the object use.
2. Yes, that wouldn't be an issue, its simply reconstructing a similar method to accept a string input.

daspecster replied:
1. ForwardGeocode() works for me.
2. is there a way to do it so you don't have to duplicate the http request code as well?

bfranco94 replied:
1, Alright, could possibly get some additional opinions on it. 
2. Yes, what do you think of handeling the HTTP request in a seperate Method, and each Forward/Reverse Geocoding method handles building the url for each request?

daspecster commented on Ziptastic/src/main/java/Ziptastic_Object/Ziptastic.java:
```diff
@@ -0,0 +1,54 @@
+        String urlBuilder = url;
+        urlBuilder = urlBuilder.concat(Integer.toString(zipcode));
+        
+        System.out.println(urlBuilder);
```
Probably want to remove this.

bfranco94 replied:
Whoops, yeah I took that out. Was using it to test to make sure my string was forming how I wanted it to.

daspecster commented on Ziptastic/src/main/java/Ziptastic_Object/Ziptastic.java:
```diff
@@ -0,0 +1,54 @@
+             
+            try (CloseableHttpResponse response = httpclient.execute(httpGet)) {
+                
+                System.out.println(response.getStatusLine());
```
Ditto on the prints.

bfranco94 replied:
Done

daspecster commented on Ziptastic/src/main/java/Ziptastic_Object/Ziptastic.java:
```diff
@@ -0,0 +1,54 @@
+                
+                System.out.println(response.getStatusLine());
+                HttpEntity entity = response.getEntity();
+                System.out.println(entity.toString());
```
last one!

bfranco94 replied:
And done

bfranco94 committed:
Removed print commands from code.

bfranco94 committed:
Renamed ForwardGeocoding -> GetFromZipcode for easier understanding of method function

bfranco94 committed:
Update README.md

bfranco94 committed:
Reverse Geocoding implemented.

Request moved into a separate method

Reverse/Forward Geocoding methods now create the URL which gets passed to the HTTP Request method.

bfranco94 committed:
Formatted code layout.

bfranco94 committed:
Delete .gitignore

bfranco94 committed:
Delete LICENSE

bfranco94 committed:
Delete README.md

bfranco94 committed:
Formatted code layout.

bfranco94 committed:
Formatted code layout.

bfranco94 committed:
Formatted code layout.

bfranco94 committed:
Created a sample TestMethod
Added Mockito and PowerMockito dependency to pom.xml

bfranco94 committed:
Added Mockito and PowerMockito dependency to pom.xml

bfranco94 committed:
Modified Dependency / Plugin

bfranco94 committed:
Created an Invalid API Key Unit Test

bfranco94 committed:
Formatted Code

bfranco94 committed:
Formatted Code

bfranco94 committed:
Removed JDK-7 dependency

daspecster commented on src/main/java/Ziptastic_Object/Ziptastic.java:
```diff
@@ -0,0 +1,65 @@
+       this.apiKey = key; 
+    }
+    
+    public static String GetFromLongLat(double latitude, double longitude, int radius) throws IOException{
```
I dig it!

daspecster commented:
Looks like there's a maven plugin for coveralls.io!
https://coveralls.zendesk.com/hc/en-us/articles/201342829-Java

bfranco94 committed:
Added Test Coverage (Coveralls)

bfranco94 committed:
Added Test Coverage (Coveralls)

bfranco94 committed:
Added Project Support for Coveralls

bfranco94 committed:
Added Project Support for Coveralls

daspecster commented:
@bfranco94 what do you think needs to be done to this in order to have it be an initial release? I was thinking forward and reverse geocoding with 100% coverage. We could merge it before the test coverage was fully implemented if you want though.

bfranco94 commented:
I don't think it should be an issue to get 100% test coverage done before we push for an Initial release. I'll probably clean up the readme file a bit as well

daspecster commented:
Here's the badge code for travis which we should add the to readme.md.

``` markdown
[![Build Status](https://travis-ci.org/Ziptastic/ziptastic-java.svg?branch=master)](https://travis-ci.org/Ziptastic/ziptastic-java)
```

Also, looks like coverage is still [86%](https://coveralls.io/github/bfranco94/ziptastic-java)?

bfranco94 committed:
Edited Travis.yml file
Added Travis.CL badged to READEME.md

bfranco94 committed:
100% Test Coverage

bfranco94 committed:
Edit for initial release

Commit:
Updated future work
